pkg/statistics/region_collection.go

```diff
@@ -428,3 +430,24 @@ func notIsolatedStoresWithLabel(stores []*core.StoreInfo, label string) [][]*cor
 	}
 	return res
 }
+
+// logDownPeerWithNoDisconnectedStore logs down peers on connected stores.
+// It won't log down peer when any store of the replica is disconnected which is
+// used to avoid too many logs when a store is disconnected.
+// TODO: it's not a good way to log down peer during process region heartbeat, we should handle it in another way.
+// region: the region which has down peer
+// stores: all stores that the region has peer on them
+func logDownPeerWithNoDisconnectedStore(region *core.RegionInfo, stores []*core.StoreInfo) {
+	for _, store := range stores {
```

Comment:
if len(stores) is 0, do we need also skip?

Reply:
It would be strange if there is a down-peer without a store, so it is necessary to log down peers in this situation I think.